### PR TITLE
Fix test-low-memory script for Mac

### PR DESCRIPTION
**Pull request by uponthesun**
I spent some time trying to get the `xargs` command to work, and got close, but it seemed fragile - there were differences between the `xargs` command on my mac and in the drone container, for example. Some people online claimed `parallel` is the better tool for this, and I saw it is available to install via `brew`, so I think it's fine to require as part of our dev setup.

All of the logic which uses `nproc`, `meminfo`, etc. to determine # of processes seems like it'll be different between Mac and Linux, so I changed the script to branch on the OS, not sure if this is a good pattern but it seems to work for my Mac, the drone container, and our managed environments. I didn't add logic yet for dynamically determining # of processes to the Mac branch since I'm not sure if it's needed. I figure getting it working for a static number is an improvement over it not working at all.

I'm also not sure whether the old memory calculations are still needed after the switch to Chrome Headless, or if they can be updated at all. My impression is that PhantomJS used more memory because of memory leaks.

## Testing story

* Tested on my mac and on drone. Strangely, `npm run lint` fails on my mac, but everything else passes, and I got a green drone run. `npm run lint` by itself fails on my mac, so that seems like a separate issue.
* Would love to have others test this version of their script on their local machines.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test-low-memory.sh [hunk 1]
```diff
@@ -1,50 +1,55 @@
...
+    local mem_metric=MemAvailable
+  else
+    local mem_metric=MemFree
+  fi
+
+  # Don't run more processes than can fit in free memory.
+  local mem_procs=$(awk "/${mem_metric}/ {printf \"%d\", \$2/1024/${MEM_PER_PROCESS}}" /proc/meminfo)
+  local procs=$(( ${mem_procs} < ${nprocs} ? ${mem_procs} : ${nprocs} ))
+
+  if [ "$PROCS" -eq 0 ]; then
```
Comment (islemaster) [inline apps/test-low-memory.sh]: I see this output when running the script:
```
./test-low-memory.sh: line 20: [: : integer expression expected
```
...although it didn't stop the script proceeding with parallelism=2.  Also `PROCS` isn't defined yet. My shell is rusty, but I think this works:

```suggestion
  if ((procs == 0)); then
```

File: apps/test-low-memory.sh [hunk 2]
```diff
@@ -1,50 +1,55 @@
...
+    local free_kb=$(awk "/MemFree/ {printf \"%d\", \$2/1024}" /proc/meminfo)
+    echo "Warning: There may not be enough free memory to run tests. Required: ${MEM_PER_PROCESS}KB; Free: ${free_kb}KB"
```
Comment (islemaster) [inline apps/test-low-memory.sh]: This `echo` inside a function messes up the return value and breaks this fallback case.

**Review by islemaster — CHANGES_REQUESTED** [review]

File: apps/test-low-memory.sh (hunk 2 above)
Comment (uponthesun) [reply apps/test-low-memory.sh]: Thanks! First time writing a function inside a shell script...

File: apps/test-low-memory.sh (hunk 1 above)
Comment (uponthesun) [reply apps/test-low-memory.sh]: Tested and it does work. thanks!

**Review by islemaster — APPROVED** [review]
This works on my machine now.  Thanks!

**Review by davidsbailey — APPROVED** [review]
My mac with 8GB RAM runs a bit slowly while running this with `PROCS=2`, but does succeed in running all the tests in under 9 minutes. This increases to 16 minutes with `PROCS=1`. I think it's fine to leave as `PROCS=2` for now, and hopefully as part of future work we can start to trim away how much memory each test run uses, since IIRC we only allocate 4GB per run to accommodate PhantomJS memory leaks.

**Pull request merged**